CMP1903M Assessment 1 Base Code/Input.cs

```diff
@@ -8,57 +8,72 @@ namespace CMP1903M_Assessment_1_Base_Code
 {
     public class Input
     {
+        private string? _input;
+        public string input { get { return (string)input; } }
         //Handles the text input for Assessment 1
 
         //Method: manualTextInput
         //Arguments: none
-        //Returns: string
+        //Returns: bool (if success or not)
         //Gets text input from the keyboard
-        public static void ManualTextInput(string Manual)
+        public bool ManualTextInput(string manual)
         {
+            Console.WriteLine("Is this correct: {0}Y/N: ", manual);
+            string? is_correct;
 
-            Console.WriteLine("\nIs this correct: {0} Y/N: ", Manual);
-            var isCorrect = Console.ReadLine();
-            isCorrect = isCorrect.ToUpper(); 
-            if (isCorrect == "Y")
+            while (true)
             {
-                Console.WriteLine("\nYou entered Yes\n");
-                Analyse.Analysis(Manual);
-            }
-            else if (isCorrect == "N")
-            {
-                Console.WriteLine("\nYou entered No\n");
-                Program.Main();
-            }
-            else if (isCorrect != "N" || isCorrect != "Y")
-            {
-                Console.WriteLine("\nNot correct");
-                Program.Main();
+                is_correct = Console.ReadLine();
+
+                if (is_correct == null)
+                {
+                    Console.WriteLine("Input is invalid, please try again: ");
+                }
+                else if (is_correct == "Y")
+                {
+                    Console.WriteLine("\nYou entered Yes\n");
+                    _input = manual;
+                    return true;
+                }
+                else if (is_correct == "N")
+                {
+                    Console.WriteLine("\nYou entered No");
+                    return false;
+                }
             }
         }   
 
         //Method: fileTextInput
         //Arguments: string (the file path)
-        //Returns: string
+        //Returns: bool (if its a success or not)
         //Gets text input from a .txt file
-        public static void FileTextInput(string File)
+        public bool FileTextInput(string path)
         {
-            
-            Console.WriteLine("Is this correct: {0}Y/N: ", File);
-            var isCorrect = Console.ReadLine();
-            isCorrect = isCorrect.ToUpper(); 
-            if (isCorrect == "Y")
-            {
-                Console.WriteLine("\nYou entered Yes\n");
-                Analyse.Analysis(File);
-            }
-            else if (isCorrect == "N")
+            var file = File.ReadAllText(path);
+
+            Console.WriteLine("Is this correct: {0}Y/N: ", file);
+            string? is_correct;
+
+            while (true)
             {
-                Console.WriteLine("\nYou entered No\n");
-                Program.Main();
+                is_correct = Console.ReadLine();
```

Comment:
We should modify this line to compensate for the removal of line 55.
We should also use the `??` operator to define a fallback value should Console.ReadLine() return `null`.
This allows us to instantiate `is_correct` as a non-optional `string`.
C# Standard recommends local variables use camelCase.
We should also use `.ToUpper()` to make the input case insensitive.
```suggestion
                string isCorrect = Console.ReadLine().ToUpper() ?? string.Empty;
```